Sources/Network/TaxonomiesService.swift

```diff
@@ -356,7 +357,28 @@ class TaxonomiesService: TaxonomiesApi {
              callback(false)
         }
     }
-
+    
+    fileprivate func refreshLabels(_ callback: @escaping (_: Bool) -> Void) {
+        do {
+            let request = try TaxonomiesRequest(route: .getLabels, requestType: .get).asURLRequest()
```

Comment:
Every TaxonomyRequest in each 'refresh' method passes .get to this method. That makes each one including this one throw an error because TaxonomiesRequest.asURLRequest() formats the path for only two matching cases:
```
private var path: String {
        switch (requestType, route) {
        case (.get, .getIngredientsAnalysisConfig):
            return route.rawValue + "/data/" + Endpoint.get
        case (.post, _):
            return Endpoint.post + route.rawValue
        default:
            return ""
        }
    }
```
So this is making it hard for me to test my code, I can't fetch the Labels. Any suggestions? What else should I know about this process?

Reply:
I do not get what you are saying. You have another case that you want to match?

Reply:
Yes, I would add 
`case (.get, _):`
to `var path`

but I only noticed that this is a problem only when the app first loads, because that is when `RootViewController` calls `taxonomiesApi.refreshTaxonomiesFromServerIfNeeded()`
At that point, none of the Taxonomies are loaded, including the new one for `Label` which I am trying to add in this PR.

So I want to give it the benefit of the doubt, that a new case _doesn't_ need to be added to `path`, because obviously the other taxonomies are working somehow, I just don't know how because I'm new to this project.

Reply:
I once added a new taxonomy, but I do not quite remember what I did, alas.

I do not see why you would want an extra case, as you always want to know the configuration parameters.

I wonder whether you need to up the model number for Realm, as you add new elements to Realm.

Reply:
That is in AppDelegate.swift.

Reply:
Or are the taxonomies not handled by Realm?

Reply:
They appear to be saved with the PersistenceManager to Realm, but I don't know whether they are fetched from Realm.

Again, I'm only going off what I've seen when debugging as soon as the app launches, so I'm sure I don't get the big picture yet. So adding an extra case appears to be the wrong thing to do.

Can you please explain to me why the Taxonomies are downloaded? I assume they aren't connected to products until a product is scanned, so it must be that at that point, a scanned product contains it's Taxonomy data, so why are the Taxonomies themselves (and all their possible values?) stored on the device?

Reply:
The taxonomies themselves are downloaded on first launch and regularly afterwards. I guess the main reason at the moment is to have an offline app. But I do not have a total insight.
By the way, I assume that you need to create a Label object to store the label data you retrieve. You can use another taxonomy as template (such as categories). However there might be some extra fields in a retrieved label object that must be added. You can check label.json to see what is sent/retrieved.

Reply:
Thanks Arnaud. Yes, that makes sense, I suspected it was for offline functionality. 
Also, I have already created a Label object to store the data, and I just copied the Category taxonomy. I will check Label.json to fill in any missing fields.

Reply:
Please see my comments on `TaxonomiesRequest` below where I explain why I made the changes discussed in this thread. I believe that none of the Taxonomies were being downloaded prior to making this change.